homeassistant/components/evohome/__init__.py

```diff
@@ -58,16 +68,57 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+ATTR_SYSTEM_MODE = "mode"
+ATTR_DURATION_DAYS = "days"
+ATTR_DURATION_HOURS = "hours"
+
+ATTR_ZONE_TEMP = "setpoint"
+ATTR_UNTIL_MINUTES = "duration"
+ATTR_UNTIL_TIME = "until"
+
+SVC_REFRESH_SYSTEM = "force_refresh"
+SVC_SET_SYSTEM_MODE = "set_system_mode"
+SVC_RESET_SYSTEM = "reset_system"
+SVC_SET_ZONE_MODE = "set_zone_override"
+SVC_RESET_ZONE_MODE = "clear_zone_override"
+
+HH_MM_REGEXP = r"^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$"  # 24h format, with leading 0
+
+SET_SYSTEM_MODE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_SYSTEM_MODE): vol.In(EVO_SYSTEM_MODES),
+        vol.Exclusive(ATTR_DURATION_DAYS, "duration"): vol.All(
```

Comment:
We usually use time period for time interval options longer than a couple of seconds. Use `cv.time_period` as the primary input format then convert that so we can validate allowed range and use it in the api. Wrap all needed validators with `vol.All`.

When the config validation is done we should be able to use the config values directly in our api.

Reply:
@MartinHjelmare I did my best - the main issue is some of the schema can't truly known until after the integration has started - e.g. some evohome systems have all possible modes, other have only a subset of them.

In addition some params (e.g. min/max temps are not fixed).

Could you have another look - if this is OK, I've got another integration to do, which is much less complicated, so I won't trouble you with it.